File: comparativa/ejemplos/07-clases/person3.py
import sys

class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    def __str__(self):
        return "Name: " + self.name + ", Age: " + str(self.age)

    def greet(cls):
        print "Hello!"

# Person.greet() => TypeError: unbound method greet() must be called with Person instance as first argument (got nothing instead)
p = Person('None',0)
p.greet()

File: comparativa/ejemplos/07-clases/person3.rb
class Person
  def initialize(name, age)
    @name = name
    @age = age
  end

  def to_s
    "Name: #{@name}, Age: #{@age}"
  end

  def self.greet
    puts("Hello!")
  end
end

Person.greet # greet is class method. Not instance method.
